django/thunderstore/api/cyberstorm/views/package_listing.py

```diff
@@ -218,3 +202,47 @@ def get_custom_package_listing(
     listing.dependant_count = get_package_dependants(listing.package.pk).count()
 
     return listing
+
+
+class PackageListingStatusAPIView(APIView):
+    permission_classes = [IsAuthenticated]
+    serializer_class = PackageListingStatusResponseSerializer
+
+    @swagger_auto_schema(
+        operation_id="cyberstorm.package_listing.status",
+        responses={200: serializer_class},
+        tags=["cyberstorm"],
+    )
+    def get(
+        self, request, namespace_id: str, package_name: str, community_id: str
+    ) -> Response:
+        package_listing = get_package_listing(
+            namespace_id=namespace_id,
+            package_name=package_name,
+            community_id=community_id,
+        )
+        checker = PermissionsChecker(package_listing, request.user)
+
+        if not (checker.can_manage or checker.can_moderate):
+            error_msg = "You do not have permission to view review information."
+            raise PermissionDenied(error_msg)
+
+        response_data = {
+            "review_status": None,
+            "rejection_reason": None,
+            "internal_notes": None,
+            "listing_admin_url": None,
+        }
+
+        if checker.can_manage:
+            response_data["review_status"] = package_listing.review_status
+            response_data["rejection_reason"] = package_listing.rejection_reason
+
+        if checker.can_view_listing_admin_page:
+            response_data["listing_admin_url"] = package_listing.get_admin_url()
+
+        if checker.can_moderate:
+            response_data["internal_notes"] = package_listing.notes
+
+        serializer = self.serializer_class(response_data)
+        return Response(serializer.data, status=status.HTTP_200_OK)
```

Comment:
Since this view is a GET view, yet returns different responses/data depending on which user accesses it, it should be moved under the current-user APIs to make it clearer it's user-bound data and shouldn't e.g. be cached publicly